### ISSUE TITLE
Support for [dedicated host types DSv3-Type3 and ESv3-Type3]

### ISSUE THREAD

**Issue by ssrirama-aka** [terraform-providers/terraform-provider-azurerm#9947]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Azure announced general availability of dedicated hosts of a new type called Type 3 in September 2020. Our team would like to transition to using these Type 3 hosts. AzureRM 2.41.0 release does not include support for Type 3 dedicated host. Running a Terraform configuration with AzureRM 2.41.0, specifying SKU DSv3-Type3 or ESv3-Type3 fails with: "Error: expected sku_name to be one of [DSv3-Type1 DSv3-Type2 ESv3-Type1 ESv3-Type2 FSv2-Type2], got DSv3-Type3"

Please add support for the dedicated host SKUs: DSv3-Type3, ESv3-Type3

### New or Affected Resource(s)

* azurerm_dedicated_host

### Potential Terraform Configuration

```hcl
resource "azurerm_dedicated_host" "testdedhost" {
  name                    = "test-dedhost-type3"
  location                = "centralus"
  dedicated_host_group_id = azurerm_dedicated_host_group.testdedhostgrp.id
  # Also this sku: sku_name = "ESv3_Type3"
  sku_name                = "DSv3-Type3"
  platform_fault_domain   = 1

      tags = {
        environment = var.environment
    }
}
```

### References

* #0000

**Comment by ruandersMSFT** [comment]
Correct, the MSFT PG did announce [new dedicated host types effective Sept 1, 2020](https://azure.microsoft.com/en-us/updates/azure-dedicated-hosts-now-support-new-azure-virtual-machines-series/).  The Dedicated Host Sku Name list is now comprised of (bold skus need to be added to bring the skus current):  

- **Dasv4_Type1**
- **Ddsv4_Type1**
- **Dsv4_Type1**
- Dsv3_Type1
- Dsv3_Type2
- **Dsv3_Type3**
- **DCsv2_Type1**
- **Easv4_Type1**
- **Edsv4_Type1**
- **Esv4_Type1**
- Esv3_Type1
- Esv3_Type2
- **Esv3_Type3**
- Fsv2_Type2
- **Fsv2_Type3**
- **Lsv2_Type1**
- **Ms_Type1**
- **Msm_Type1**
- **Msmv2_Type1**
- **Msv2_Type1**
- **NVasv4_Type1**
- **NVsv3_Type1**

**Comment by ruandersMSFT** [comment]
PR #9949 ready.

**Comment by ssrirama-aka** [comment]
@ruandersMSFT I took PR [#9951](https://github.com/terraform-providers/terraform-provider-azurerm/pull/9951) and built the provider locally. Tested for DSv3-Type3 host in my subscription. All worked cleanly: plan, apply, destroy

**Comment by katbyte** [comment]
Closing as it looks like this was fixed by #9951
